src/udev/udev-builtin-kmod.c

```diff
@@ -56,31 +52,35 @@ static int builtin_kmod(UdevEvent *event, int argc, char *argv[]) {
 
 /* called at udev startup and reload */
 static int builtin_kmod_init(void) {
+        int r;
+
         if (ctx)
                 return 0;
 
-        ctx = kmod_new(NULL, NULL);
-        if (!ctx)
-                return -ENOMEM;
-
         log_debug("Loading kernel module index.");
```

Comment:
This should be moved down?

Reply:
hmm, where to?

Reply:
Previously this is logged only if `kmod_new` succeeds, so this should stay below `module_setup_context`?

Reply:
sure, but that's a minor detail. the kmod_new() is a cheap call, that just malloc()s a bit, and never really fails. the heavy call is kmod_load_resources(), which loads stuff of disk. I think the debug message should tell us that this heavy operation is now initiating, hence the log message should be *before* it. The fact that it is now also before kmod_new() doesn't really matter much I'd say.

Reply:
OK. That makes sense.